### DISCUSSION TITLE
GUI element imported too early in Arch.makeWindow

### DISCUSSION THREAD

**Discussion by rix133** [FreeCAD/FreeCAD#17598] [Code contribution discussions]
The GUI component is imported before needed [from DraftGui import todo](https://github.com/FreeCAD/FreeCAD/blame/8865450a3e14220925e0e449c0f1f79056b4fb89/src/Mod/BIM/Arch.py#L1089)   so if run without GUI one will get an error 

A solution would be to add this import after the GUI check eg after [if obj.Base and FreeCAD.GuiUp:](https://github.com/FreeCAD/FreeCAD/blame/8865450a3e14220925e0e449c0f1f79056b4fb89/src/Mod/BIM/Arch.py#L1137)

**Comment by luzpaz** [comment]
Transferred to https://github.com/FreeCAD/FreeCAD/issues/17600  
Closing in favor of newly created ticket.
